diff --git a/neutron/templates/bin/_neutron-ovn-init.sh.tpl b/neutron/templates/bin/_neutron-ovn-init.sh.tpl
index e6571a3..4937a53 100644
--- a/neutron/templates/bin/_neutron-ovn-init.sh.tpl
+++ b/neutron/templates/bin/_neutron-ovn-init.sh.tpl
@@ -17,10 +17,11 @@ limitations under the License.
 set -ex
 
 # See: https://bugs.launchpad.net/neutron/+bug/2028442
+{{- $ovn := default dict .Values.conf.plugins.ml2_conf.ovn }}
 mkdir -p /tmp/pod-shared
 tee > /tmp/pod-shared/ovn.ini << EOF
 [ovn]
-ovn_nb_connection={{ coalesce .Values.conf.plugins.ml2_conf.ovn.ovn_nb_connection "tcp:$OVN_OVSDB_NB_SERVICE_HOST:$OVN_OVSDB_NB_SERVICE_PORT_OVSDB" }}
-ovn_sb_connection={{ coalesce .Values.conf.plugins.ml2_conf.ovn.ovn_sb_connection "tcp:$OVN_OVSDB_SB_SERVICE_HOST:$OVN_OVSDB_SB_SERVICE_PORT_OVSDB" }}
+ovn_nb_connection={{ coalesce $ovn.ovn_nb_connection "tcp:$OVN_OVSDB_NB_SERVICE_HOST:$OVN_OVSDB_NB_SERVICE_PORT_OVSDB" }}
+ovn_sb_connection={{ coalesce $ovn.ovn_sb_connection "tcp:$OVN_OVSDB_SB_SERVICE_HOST:$OVN_OVSDB_SB_SERVICE_PORT_OVSDB" }}
 EOF
