diff --git a/ovn/templates/bin/_ovn-controller-init.sh.tpl b/ovn/templates/bin/_ovn-controller-init.sh.tpl
index 2780489b..1d303c8d 100644
--- a/ovn/templates/bin/_ovn-controller-init.sh.tpl
+++ b/ovn/templates/bin/_ovn-controller-init.sh.tpl
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ANNOTATION_KEY="openstack-helm/ovn-system-id"
+ANNOTATION_KEY="atmosphere.cloud/ovn-system-id"
 
 function get_ip_address_from_interface {
   local interface=$1
@@ -82,7 +82,7 @@ function get_current_system_id {
 }
 
 function get_stored_system_id {
-  kubectl get node "$NODE_NAME" -o "jsonpath={.metadata.annotations.openstack-helm/ovn-system-id}"
+  kubectl get node "$NODE_NAME" -o "jsonpath={.metadata.annotations.atmosphere\.cloud/ovn-system-id}"
 }
 
 function store_system_id() {
