# Note: Not commit to openstack_helm yet
diff --git a/ovn/templates/bin/_ovn-bgp-agent-init.sh.tpl b/ovn/templates/bin/_ovn-bgp-agent-init.sh.tpl
index 45589c66..bdf767ec 100644
--- a/ovn/templates/bin/_ovn-bgp-agent-init.sh.tpl
+++ b/ovn/templates/bin/_ovn-bgp-agent-init.sh.tpl
@@ -24,9 +24,15 @@ ovn_nb_connection=tcp:$OVN_OVSDB_NB_SERVICE_HOST:$OVN_OVSDB_NB_SERVICE_PORT_OVSD
 ovn_sb_connection=tcp:$OVN_OVSDB_SB_SERVICE_HOST:$OVN_OVSDB_SB_SERVICE_PORT_OVSDB
 EOF
 
-{{- if empty .Values.conf.ovn_bgp_agent.DEFAULT.bgp_router_id }}
+{{- if or (empty .Values.conf.ovn_bgp_agent.DEFAULT.bgp_router_id) (empty .Values.conf.ovn_bgp_agent.frr_k8s.node_name) }}
 tee > /tmp/pod-shared/ovn-bgp-agent.ini << EOF
+{{- if empty .Values.conf.ovn_bgp_agent.DEFAULT.bgp_router_id }}
 [DEFAULT]
 bgp_router_id=$NODE_IP
+{{- end }}
+{{- if empty .Values.conf.ovn_bgp_agent.frr_k8s.node_name }}
+[frr_k8s]
+node_name=$NODE_NAME
+{{- end }}
 EOF
 {{- end }}
diff --git a/ovn/templates/daemonset-ovn-bgp-agent.yaml b/ovn/templates/daemonset-ovn-bgp-agent.yaml
index b6525d4b..b692249a 100644
--- a/ovn/templates/daemonset-ovn-bgp-agent.yaml
+++ b/ovn/templates/daemonset-ovn-bgp-agent.yaml
@@ -61,6 +61,10 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: status.podIP
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
           volumeMounts:
           - name: pod-tmp
             mountPath: /tmp
