diff --git b/horizon/templates/bin/_django.wsgi.tpl a/charts/horizon/templates/bin/_django.wsgi.tpl
index 4626f21a..cbf87fbf 100644
--- b/horizon/templates/bin/_django.wsgi.tpl
+++ a/horizon/templates/bin/_django.wsgi.tpl
@@ -22,8 +22,6 @@ import os
 import sys
 
 import pymysql
-
-pymysql.version_info = (1, 4, 0, "final", 0)
 pymysql.install_as_MySQLdb()
 
 from django.core.wsgi import get_wsgi_application
diff --git b/horizon/templates/bin/_manage.py.tpl a/charts/horizon/templates/bin/_manage.py.tpl
index 105a32c8..1b21cfc9 100644
--- b/horizon/templates/bin/_manage.py.tpl
+++ a/horizon/templates/bin/_manage.py.tpl
@@ -23,7 +23,6 @@ import os
 import sys
 
 import pymysql
-pymysql.version_info = (1, 4, 0, "final", 0)
 pymysql.install_as_MySQLdb()
 
 from django.core.management import execute_from_command_line
